### ISSUE TITLE
[FEATURE] Configuration Option for @ToString to print in JSON format

### ISSUE THREAD

**Issue by mjohns39** [rzwitserloot/lombok#2449]
**Describe the feature**
The ability to configure the `@ToString` to output in JSON format as opposed to the default "class name, along with each field, in order, separated by commas."

Ideally, you could configure this setting on both a class level (via @ToString and @Data) annotations and on a global level (via lombok.config).

Local Examples:
```
@ToString(format = ToStringFormat.JSON)
public class POJO {

}
```

```
@Data(format = ToStringFormat.JSON)
public class POJO {

}
```

Global Example:

`lombok.config`
```
lombok.toStringFormat = JSON
```

**Describe the target audience**
JSON is a popular format for REST API.  I constantly find myself logging a POJO in json format so I can easily debug requests.  
It would be nice to do something like:
```
log.debug("POJO:  {}", pojo.toString()); //pojo.toString() prints in JSON format
```

As opposed to something like this:

```
log.debug("POJO:  {}", new Gson().toJson(pojo));
```

**Additional context**
Obviously, this is just a nice to have feature.  I only want it because of debugging and logging, but if I had to guess, there's probably some other devs out there that would like Lombok ToString to print in JSON format (and possibly other formats) for various reasons.  It would be nice to have some out of the box configuration options for the ToString method in general.

**Comment by Maaartinus** [comment]
The problem is that Lombok can't do it. It could change the format of the output, so it looks JSON-like, but when it delegates to another `.toString()`, things can break. I'm afraid, it can't do any better than generating
```
public String toString() {return new Gson().toJson(this));}
```
and that's something the user can do themselves in a much more flexible way.

Moreover, JSON lacks the information about the class, which is sometimes important for debugging.

**Comment by jm98** [comment]
The best solution that follows SOLID principals and Functional Programming would be to use Lombok @ExtensionMethods and create a private static <T> String toJson(T source) and implement it using a JSON formatter of your choice

**Comment by RezaGholamzad** [comment]
hi.
If we have a general config that we can say that all toStrings are generated in json format, then there will be no more inconsistencies.
Need to do this because we want to use tools like logstash and their json parser.
What is the opinion of the Lombok team in this regard?

**Comment by janrieke** [comment]
No, that's still insufficient: `toString()` for classes like maps or sets does not work, as well as all classes without `@ToString`. There is simply no way to do this.

**Comment by dstango** [comment]
If it would be feasible I'd prefer a generic config like

```
toString=new Gson().toJson(this));
```

... or whatever command a developer might prefer, instead of making lombok hard-coded dependent on whichever specific other library.

This would give maximum customisation while keeping lombok's effort and maintenance burden at a minimum.

Prerequisite: lombok's annotation processing can do such a thing easily.

**Comment by exabrial** [comment]
I'm completely fine with "If it delegates to another toString that isn't JSON, then it breaks". This is the way Apache Common's JSON Reflection JSON toString() works and it's still very useful.

**Comment by janrieke** [comment]
It may be fine for you, but it will obviously create huge expectations that can never be fulfilled by Lombok due to technical reasons already explained here.

You won't have to answer the incoming pile of bug reports and feature requested. The Lombok team has to.

And don't just use performance as an argument again and without plausible measurements. I doubt that a runtime-optimized Jackson serialization is much slower than a native implementation.

**Comment by s-wangc** [comment]
This is really important, the JSON format is more versatile. In fact, it is very simple to achieve this. There is no need to use libraries such as Gosn and Jackson. Lombok only needs to replace = with: (replace with {,) with } when outputting toString. This seems very simple, why doesn't lombok do this?
Or lombok provides a way for users to configure the output symbols of attributes in toString. The output symbols of objects can be configured by users themselves using =, (,) or: ,{,}.

**Comment by janrieke** [comment]
Because it is _not_ simple, as explained above. Use a JSON library, because lombok will not implement it.
